scripts/gtisc/docker_build/Dockerfile

```diff
@@ -1,35 +1,50 @@
-FROM ubuntu:22.04
+# Base image: needed for both building and running
+FROM ubuntu:22.04 as base
 
 ARG DEBIAN_FRONTEND=noninteractive
 ENV TZ=Etc/UTC
 
-RUN apt update
-RUN apt install -y --fix-missing qemu qemu-kvm
+RUN apt update && apt install -y --fix-missing qemu qemu-kvm git curl wget lsb-release sudo gcc python3 python3-dev python3-venv python3-pip python-is-python3
 
-RUN apt install -y git curl wget lsb-release sudo gcc python3 python3-dev python3-venv python3-pip
 RUN useradd -m s2e && echo "s2e:s2e" | chpasswd && adduser s2e sudo
 RUN echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
-
-RUN apt install python-is-python3
-
 USER s2e
-WORKDIR /home/s2e
 
-RUN git clone https://github.com/GTISC/s2e-env.git
-WORKDIR /home/s2e/s2e-env
+# install s2e-env for building and running s2e
+RUN cd /home/s2e && git clone https://github.com/GTISC/s2e-env.git \
+    && cd /home/s2e/s2e-env && pip install --upgrade pip && pip install .
 
 ENV PATH=$PATH:/home/s2e/.local/bin
-RUN pip install --upgrade pip
-RUN pip install .
+
+# ================
+# First stage: build s2e
+FROM base as builder
 
 ARG TARGET_BRANCH=master
 RUN s2e init /home/s2e/s2e -sb $TARGET_BRANCH
-
 WORKDIR /home/s2e/s2e
 SHELL ["/bin/bash", "-c"]
 RUN source s2e_activate
 RUN s2e update
 RUN s2e build
-ENV TERM=xterm-256color
```

Comment:
This env is necessary to start S2E in the headless mode.

Reply:
LMAO, I just got an error and was wondering where was wrong. Let me add it back and test again.